src/main/scala/org/ergoplatform/nodeView/history/ErgoHistory.scala

```diff
@@ -185,6 +184,35 @@ trait ErgoHistory
     case _ => None
   }
 
+  /**
+    * Remove header, corresponding block parts, and corresponding indexes from storage and caches
+    * @param headerId - header id
+    * @return
+    */
+  def forgetHeader(headerId: ModifierId): Try[Unit] = Try {
+    typedModifierById[Header](headerId).foreach { h =>
+      historyStorage.remove(
+        Seq(
+          validityKey(headerId),
+          headerHeightKey(headerId),
+          headerScoreKey(headerId)
+        ),
+        Seq(
+          headerId
+        )
+      ).get
+      requiredModifiersForHeader(h).foreach { case (_, mId) =>
+        historyStorage.remove(
+          Seq(
+            validityKey(mId)
+          ),
+          Seq(
+            mId
+          )
+        ).get
```

Comment:
Kind of weird formatting :). Let's maybe use named arguments for clarity.

Reply:
fixed